File: ohara-kafka/src/test/scala/com/island/ohara/kafka/connector/SimpleRowSinkConnector.scala
package com.island.ohara.kafka.connector

import Constants._
import com.island.ohara.util.VersionUtil

/**
  * Used for testing.
  */
class SimpleRowSinkConnector extends RowSinkConnector {
  private[this] var config: TaskConfig = _
  override val _version = VersionUtil.VERSION

  override def _start(props: TaskConfig): Unit = {
    this.config = props
    // check the option
    this.config.options(OUTPUT)
    this.config.options(BROKER)
  }

  override def _taskClass(): Class[_ <: RowSinkTask] = classOf[SimpleRowSinkTask]

  override def _taskConfigs(maxTasks: Int): Seq[TaskConfig] = Seq.fill(maxTasks)(config)

  override def _stop(): Unit = {}
}

File: ohara-kafka/src/test/java/com/island/ohara/kafka/connector/SimpleRowSinkConnector.java
package com.island.ohara.kafka.connector;

import static com.island.ohara.kafka.connector.Constants.BROKER;
import static com.island.ohara.kafka.connector.Constants.OUTPUT;

import java.util.Collections;
import java.util.List;

/** Used for testing. */
public class SimpleRowSinkConnector extends RowSinkConnector {
  @Override
  protected void _start(TaskConfig props) {
    this.config = props;
    // check the option
    this.config.options().get(OUTPUT);
    this.config.options().get(BROKER);
  }

  @Override
  protected void _stop() {}

  @Override
  protected Class<? extends RowSinkTask> _taskClass() {

    return SimpleRowSinkTask.class;
  }

  @Override
  protected List<TaskConfig> _taskConfigs(int maxTasks) {
    return Collections.nCopies(maxTasks, config);
  }

  private TaskConfig config = null;
}
